hibernate-core/src/main/java/org/hibernate/query/criteria/internal/predicate/CompoundPredicate.java

```diff
@@ -125,6 +125,13 @@ public Predicate not() {
 		return new NegatedPredicateWrapper( this );
 	}
 
+	public void wrap() {
+		final CriteriaBuilderImpl criteriaBuilder = criteriaBuilder();
+		for ( int i = 0, size = expressions.size(); i < size; i++ ) {
+			expressions.set( i, criteriaBuilder.wrap( expressions.get( i ) ) );
```

Comment:
Maybe it's better to do the wrapping in `applyExpressions` as that is the only code where expressions are added to the list.

Reply:
done. thanks.